packages/base-map/src/index.js

```diff
@@ -0,0 +1,284 @@
+    React.Children.toArray(children).forEach(child => {
+      if (child.props.name) userControlledOverlays.push(child);
+      else fixedOverlays.push(child);
+    });
+
+    return (
+      <Map
+        ref="map"
+        className="map"
+        center={center}
+        zoom={zoom}
+        maxZoom={maxZoom}
+        // onClick={this.onLeftClick}
+        // Note: Map-click is handled via single-click plugin, set up in componentDidMount()
+        onOverlayAdd={onOverlayAdded}
+        onOverlayRemove={onOverlayRemoved}
+        onViewportChanged={onViewportChanged}
+      >
+        {/* Create the layers control, including base map layers and any
+         * user-controlled overlays. */}
+        <LayersControl position="topright">
+          {/* base layers */}
+          {baseLayers &&
+            baseLayers.map((layer, i) => {
+              return (
+                <LayersControl.BaseLayer
+                  name={layer.name}
+                  checked={i === 0}
+                  key={i}
+                >
+                  {
+                    // Fix tile size/zoom offset: https://stackoverflow.com/a/37043490/915811
+                    // Also, split the declaration between the two
+                    // outcomes to avoid error 'attempted to load an infinite number of tiles'
+                    // if the tiles cannot be loaded for any reason.
+                  }
+                  {L.Browser.retina && layer.hasRetinaSupport ? (
+                    <TileLayer
+                      url={layer.url}
+                      attribution={layer.attribution}
+                      maxZoom={layer.maxZoom}
+                      tileSize={512}
+                      zoomOffset={-1}
+                      detectRetina
+                    />
+                  ) : (
+                    <TileLayer
+                      url={layer.url}
+                      attribution={layer.attribution}
+                      maxZoom={layer.maxZoom}
+                      detectRetina
+                    />
+                  )}
+                </LayersControl.BaseLayer>
+              );
+            })}
+
+          {/* user-controlled overlay layers (e.g., vehicle locations, stops) */}
+          {userControlledOverlays.map((child, i) => (
+            <LayersControl.Overlay
+              key={i}
+              name={child.props.name}
+              checked={child.props.visible}
+            >
+              {child}
+            </LayersControl.Overlay>
+          ))}
+        </LayersControl>
+
+        {/* Add the fixed, i.e. non-user-controllable, overlays (e.g., itinerary overlay) */}
+        {fixedOverlays}
+
+        {/* Add the location selection popup and content, if both are set. */}
+        {popupLocation && popupContent && (
+          <Popup position={popupLocation} onClose={onPopupClosed}>
+            {popupContent}
+          </Popup>
+        )}
+      </Map>
+    );
+  }
+}
+
+const twoNumberArrayChecker = (
```

Comment:
Since we're going all out with a custom PropType here, we should also include checks for whether the values are valid GPS coordinates. IE -90 <= lat <= 90 and -180 <= lon <= 180.

Reply:
Good point

Reply:
Might have gone too far in the latest commit with the props validation... Let me know if you want to go back.

Reply:
I like it. One thing tho, I'd move the props validation into this new file I created: `core-utils/src/types.js`.

Reply:
Should I merge with the itinerary branch to get that file?

Reply:
I merged with add-itinerary-eas to get the most current outside of master.